engine/namegen.ml

```diff
@@ -383,19 +383,24 @@ let next_name_away_with_default_using_types default na avoid t =
 
 let next_name_away = next_name_away_with_default default_non_dependent_string
 
+let make_all_rel_context_name_different env sigma ctx =
+  let avoid = ref (Id.Set.union (Context.Rel.to_vars (Environ.rel_context env)) (ids_of_named_context_val (named_context_val env))) in
+  Context.Rel.fold_outside
+    (fun decl (newenv,ctx) ->
+       let na = named_hd newenv sigma (RelDecl.get_type decl) (RelDecl.get_name decl) in
+       let id = next_name_away na !avoid in
+       avoid := Id.Set.add id !avoid;
+       let decl = RelDecl.set_name (Name id) decl in
+       push_rel decl newenv, decl :: ctx)
+    ctx ~init:(env,[])
+
 let make_all_name_different env sigma =
   (* FIXME: this is inefficient, but only used in printing *)
```

Comment:
What is the inefficient bit?

Reply:
The comment was from @ppedrot. Maybe from the time `avoid` was a list. (If so, the comment can be removed.)